### ISSUE TITLE
A comma (,) in irc.look.highlight_channel does not work as expected

### ISSUE THREAD

**Issue by quite** [weechat/weechat#512]
I'm trying to set highlight_channel to: ```"$nick:", "$nick,"```

But this does not work. It seems to be interpreted as the empty string, and highlighting is thus disabled. Yes, I'l like to have the highlight for my nick (quite) just like this ;)

**Comment by quite** [comment]
I realize that the actual string that highlight_channel is ```"$nick:,$nick,```
farn suggested escaping the (non-separating) comma with a "\" but that did not do it.

**Comment by flashcode** [comment]
Hi,
Can you please tell me the exact command you typed, the result and expected result (and your WeeChat version). Then I'll try to reproduce the problem, it will be easier to fix.
Thanks.

**Comment by quite** [comment]
Perhaps, the title is misleading. It's of course how I expected it to work, which may or may not be sound....

I tried the following, in expecting to have highlight when somebody types "quite:" or "quite,"

    /set irc.look.highlight_channel "$nick:,$nick,"

But, I realised that it might be impossible for the parser to guess how I would like that interpreted. What the above accomplishes is, quite correctly I'd say now, highlighting on "quite" (and "quite:", which is of course already covered).

In short, what I would like is a way to let the comma be part of the highlight string, thus there is a need for somehow being able to quote the comma, so the parser just won't see it as a string separator.

**Comment by flashcode** [comment]
I see. Then this is more a new feature you're asking for.
In many options, there's a split on comma or semicolon, and this separator can not be part of resulting strings.
This would require some changes to allow at least an escape of the separator char (but maybe not everywhere).
